### PR TITLE
#10279: Show credits/attribution text in printed page

### PR DESCRIPTION
**Pull request by mahmoudadel54**
## Description
In this PR, adding credit/attribution section to print map is enabled by adding a new section into config.yaml of print.
Now in print map, the attribution section will be displayed if the map has layers with credit titles [attributions]. 
Update the config.yaml file is required if attribution/credit section is need to added. 
This PR includes also migration guide for this.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
#10279 

**What is the current behavior?**
#10279 

**What is the new behavior?**
The credits/attributions text will be displayed in printed page.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/utils/PrintUtils.js [hunk 1]
```diff
@@ -229,6 +229,45 @@ export const getMapSize = (layout, maxWidth) => {
 
 export const mapProjectionSelector = (state) => state?.print?.map?.projection ?? "EPSG:3857";
 
+export function parseCreditRemovingTagsOrSymbol(creditText = "") {
```
Comment (MV88) [inline web/client/utils/PrintUtils.js:232]: remember always to add js docs especially for function like this

File: web/client/utils/PrintUtils.js [hunk 2]
```diff
@@ -229,6 +229,45 @@ export const getMapSize = (layout, maxWidth) => {
...
+            const hasHtmlTag = layerCredit.includes('<');
+            if (hasHtmlTag || hasOrSymbol) {
+                layerCredit = parseCreditRemovingTagsOrSymbol(layerCredit);
+            }
+            cumCredit = cumCredit ? cumCredit + " | " + layerCredit : layerCredit;
+        }
+        return cumCredit;
+    }, '');
+    return credits;
+};
```
Comment (MV88) [inline web/client/utils/PrintUtils.js:269]: your version is a bit cumbersome and can be simplified this way
```suggestion
export const getLayersCredits = (layers) => {
    const layerCredits = layers.map((layer) => {
        const layerCreditTitle = layer?.credits?.title || '';
        const hasOrSymbol = layerCreditTitle.includes('|');
        const hasHtmlTag = layerCreditTitle.includes('<');
        const layerCredit = (hasHtmlTag || hasOrSymbol)
            ? parseCreditRemovingTagsOrSymbol(layerCreditTitle)
            : layerCreditTitle;
        return layerCredit;
    }).join(' | ');
    return layerCredits;
};

```

**Review by MV88 — CHANGES_REQUESTED** [review]
@mahmoudadel54 
Please checkout my comments.
There are needed a few changes here and there

File: web/client/utils/PrintUtils.js (hunk 2 above)
Comment (mahmoudadel54) [reply web/client/utils/PrintUtils.js:269]: Nice way, thanks

File: docs/developer-guide/mapstore-migration-guide.md [hunk 3]
```diff
@@ -22,6 +22,33 @@ This is a list of things to check if you want to update from a previous version
 
 ## Migration from 2024.01.00 to 2024.01.02
 
+### Enable showing credits/attribution text in Print config
+
+Due to showing layers' credits/attributions of printed map which will be displayed at the bottom of the map section, the MapStore `config.yaml` file should be reviewed and updated. Below are reported the relevant changes that need to be applied also to `config.yaml` of MapStore donwstream projects where the printing engine is present.
```
Comment (MV88) [inline docs/developer-guide/mapstore-migration-guide.md:27]:
```suggestion
Due to showing layers' credits/attributions of printed map which will be displayed at the bottom of the map section, the MapStore `config.yaml` file should be reviewed and updated. Below are reported the relevant changes that need to be applied also to `config.yaml` of MapStore downstream projects where the printing engine is present.
```

**Review by MV88 — APPROVED** [review]

**Review by MV88 — APPROVED** [review]

**Comment by MV88** [comment]
@ElenaGallo please test it in DEV

**Comment by ElenaGallo** [comment]
Test passed, @mahmoudadel54 please backport to 2024.01.xx. Thanks

**Pull request merged**